homeassistant/components/sensor/imap_email_content.py

```diff
@@ -232,6 +238,15 @@ def update(self):
                     EmailContentSensor.get_msg_subject(email_message),
                 ATTR_DATE:
                     email_message['Date'],
-                ATTR_BODY:
-                    EmailContentSensor.get_msg_text(email_message)
             }
+
+            if self._body_regex is not None:
+                parsed_body = self._body_regex.search(
+                    EmailContentSensor.get_msg_text(email_message))
+                if parsed_body is not None:
+                    for attr in parsed_body.groupdict().keys():
+                        self._state_attributes[attr] = parsed_body.\
```

Comment:
I'm not sure that the names of the groups should just be used as keys here. What if a user has the bright idea of calling his group `last_updated`? Or `friendly_name`? I'm not sure in which subtle way this will break HA. How about instead having some `ATTR_PARSED_BODY` or something, and then set `self._state_attributes[ATTR_PARSED_BODY] = {attr : parsed_body.group(attr) for attr in parsed_body.groupdict().keys()}`